src/app/static/src/app/utils.ts

```diff
@@ -37,33 +38,44 @@ export const mapStatePropByName = <T>(namespace: string | null, name: string): C
     return (namespace && mapState(namespace, [name])[name]) || mapState([name])[name]
 };
 
-export const mapStateProps = <S, K extends string>(namespace: string,
-    map: Dict<(this: CustomVue, state: S) => any>) => {
-    type R = {[key in K]: any}
-    return mapState<S>(namespace, map) as R
+export const mapStateProps = <M extends Dict<(this: CustomVue, state: any) => any>>(namespace: string, map: M) => {
+    type T = typeof map
+    type R = {[key in keyof T]: ComputedGetter<ReturnType<T[key]>>}
+    return mapState(namespace, map) as R
 };
 
-export const mapGetterByName = <T>(namespace: string | null, name: string): ComputedWithType<T> => {
+export const mapRootStateProps = <M extends Dict<(this: CustomVue, state: any) => any>>(map: M) => {
+    type T = typeof map
+    type R = {[key in keyof T]: ComputedGetter<ReturnType<T[key]>>}
+    return mapState(map) as R
+};
+
+export const mapGetterByName = <T>(namespace: string | null, name: string): ComputedGetter<T> => {
     return (namespace && mapGetters(namespace, [name])[name]) || mapGetters([name])[name]
 }
 
-export const mapGettersByNames = <K extends string>(namespace: string, names: string[]) => {
-    type R = {[key in K]: any}
-    return mapGetters(namespace, names) as R
+export const mapGettersByNames = <N extends readonly string[], T extends Record<N[number], any>>(namespace: string, names: N) => {
+    const nameCopy = [...names]
+    type R = {[key in N[number]]: any}
+    type TR = {[key in N[number]]: ComputedGetter<T[key]>}
+    type Result = T extends undefined ? R : TR
+    return mapGetters(namespace, nameCopy) as Result
 };
 
 export const mapActionByName = <T>(namespace: string | null, name: string): ActionMethod => {
     return (!!namespace && mapActions(namespace, [name])[name]) || mapActions([name])[name]
 };
 
-export const mapActionsByNames = <K extends string>(namespace: string | null, names: string[]) => {
-    type R = {[key in K]: any}
-    return (!!namespace && mapActions(namespace, names) || mapActions(names)) as R
+export const mapActionsByNames = <N extends readonly string[]>(namespace: string | null, names: N) => {
+    const nameCopy = [...names]
+    type R = {[key in N[number]]: any}
+    return (!!namespace && mapActions(namespace, nameCopy) || mapActions(nameCopy)) as R
 };
 
-export const mapMutationsByNames = <K extends string>(namespace: string, names: string[]) => {
-    type R = {[key in K]: any}
-    return mapMutations(namespace, names) as R
+export const mapMutationsByNames = <N extends readonly string[]>(namespace: string, names: N) => {
+    const nameCopy = [...names]
+    type R = {[key in N[number]]: any}
+    return mapMutations(namespace, nameCopy) as R
```

Comment:
can you allow namespace to accept null value ?

Reply:
again can put this to another ticket, for now just want to focus on types and the old `mapMutationsByNames` only allows strings too